docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,524 @@ overview page.
+
+Loki Storage
+____________
+
+Since we're using filesystem to store Loki's data, we have basically 2 ways
+for having multiple Loki instances running at the same time, with the same
+set of data.
+
+Either we do like Prometheus and we store everything on every instance of
+Loki, but it means we raise the storage, RAM and CPU needs for each additionnal
+instance, either we use a `new experimental feature`_ from Loki 1.5.0 where
+Loki ingesters use a hash ring and talk between them to route the queries to
+the right one. This approach needs an external KV store to work such as
+etcd or Consul.
+
+We choose to use the first approach as the second is not production ready
+and since we plan to only do short term retention on Loki, the impact on
+storage will not be that much important.
+Moreover, it eases the deployment and maintenance since there is not an
+extra component.
+
+.. _new experimental feature: https://github.com/grafana/loki/blob/master/docs/operations/storage/filesystem.md#new-and-very-experimental-in-150-horizontal-scaling-of-the-filesystem-store
+
+.. todo::
+
+  define loki storage sizing depending on retention, ingestion rate,
+  average size of logs, number of streams, ...
+
+Configuration
+~~~~~~~~~~~~~
+
+Fluent Bit
+""""""""""
+
+Fluent Bit needs to be configured to scrape and handle properly journal and
+containers logs by default.
+
+For containers logs, we want to add the following labels:
+
+ - node: the node it comes from
+ - namespace: the namespace the pod is running in
+ - instance: the name of the pod
+ - container: the name of the container
+
+For journal we want these labels:
+
+ - node: the node it comes from
+ - unit: the name of the unit generating these logs
+
+This configuration will also be customizable by the user to be able to
+add new routes (Output) to push the log streams to.
+
+This configuration will be done through various CRs provided by the Fluent Bit
+Operator:
+
+ - FluentBit: Defines Fluent Bit instances and its associated config
+ - FluentBitConfig: Select input/filter/output plugins and generates the final
+   config into a Secret
+ - Input: Defines input config sections
+ - Filter: Defines filter config sections
+ - Output: Defines output config sections
+
+For example, if a user wants to forward all Kubernetes logs to an external log
+centralization system (e.g. Elasticisearch), he will need to define an
+``Output`` CR as follows:
+
+.. code-block:: yaml
+
+  kind: Output
+  metadata:
+    name: my-output-to-external-es
+    namespace: my-namespace
+  spec:
+    match: kube.*
+    es:
+      host: 10.0.0.1
+      port: 9200
+
+More details can be found on `Fluent Bit Operator repository`_ and manifest
+samples are
+`here <https://github.com/kubesphere/fluentbit-operator/tree/master/manifests/logging-stack>`_.
+
+.. _Fluent Bit Operator repository: https://github.com/kubesphere/fluentbit-operator
+
+Loki
+""""
+
+Loki's configuration is stored as a Secret, as for Fluent Bit, we need
```

Comment:
Minor: Since we use fluent-bit operator it's not "as for Fluent Bit" I think

Reply:
Yes, good catch, it's a leftover from the previous version of the design.